### ISSUE TITLE
✅ 🐞 Bug Report: Swiper adds `inert` incorrectly on slide - [ED-11434]

### ISSUE THREAD

**Issue by perks98** [elementor/elementor#23039]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Two instances I found this issue, one where I was trying to create a plugin that creates a carousel widget to do a certain purpose, and the main time for two client websites when using the new carousel widget.

The issue is when you partially slide to another slide panel but stop so you remain on that slide, it locks the slide from being interactive as it incorrectly adds `inert` to the HTML so you can't drag your mouse to scroll or interact with buttons on the slide. This is primarily an issue with Swiper v8.45 while v5.36 has a similar issue but the difference is you can still drag the slide in v5.36 but can't click on anything inside the slide.

### Steps to reproduce

1. Make sure Elementor and Elementor Pro is enabled on a WordPress install.
2. In Elementor Settings page make sure "Upgrade Swiper Library", "Flexbox Container" and "Nested Elements" are set to "Active".
3. Create a new page labelled how you want e.g. "Test Page No.1" then edit the page with Elementor.
4. Add the new Carousel widget to the page and for ease set the "Slides on display to "1".
5. Create 3 slides and add a heading widget and button link widget to each slide. Also give each slider its own background colour or background image to tell the difference between them.
6. Publish your changes to the page and view this page on the frontend of the website (issue still works in the editor as well).
7. Once the page first loads on the frontend use your mouse to drag the slide going to the left slide so you are now on slide 3.
8. Now use your mouse to drag back to slide 1 but don't actually go there only transition mid-way then go back to slide 3.
9. Now attempt to slide again using your mouse to drag or interact with the button, you will notice that you can't. Make sure to do this attempt right away as the automatic transition will fix itself and using the arrows or pagination will also fix itself but not ideal for clients.

**Quick demo video:**

https://github.com/elementor/elementor/assets/55829113/09a5f684-a0ab-41f8-bf34-433a669619ec

![Screenshot_825](https://github.com/elementor/elementor/assets/55829113/a5943788-ab9a-4b4c-aa6e-636abc48c29d)

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to view</summary>
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MariaDB Server v10.4.26
	PHP Version: 7.4.33
	PHP Memory Limit: 128M
	PHP Max Input Vars: 2500
	PHP Max Post Size: 128M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2.2
	Site URL: [private]
	Home URL: [private]
	WP Multisite: Yes
	Max Upload Size: 128 MB
	Memory limit: 2048M
	Max Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: en-GB
	Timezone: Europe/London
	Admin Email: [private]
	Debug Mode: Active

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.8.1
	Parent Theme Author: Elementor Team

== User ==
	Role: 
	WP Profile lang: en_GB
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor Pro
		Version: 3.15.0-beta1
		Author: Elementor.com

	WooCommerce
		Version: 7.4.0
		Author: Automattic

== Network Plugins ==
	Elementor
		Version: 3.15.0-beta2
		Author: Elementor.com

== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	woocommerce: Active

== Elementor Experiments ==
	Optimised DOM Output: Active by default
	Improved asset loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active
	Grid Container: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Editor Top Bar: Inactive by default
	Landing pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Global Style Guide: Inactive by default
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Inactive by default
	Taxonomy Filter: Inactive by default

== Log ==

Log: showing 10 of 102023-07-10 09:13:24 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.14.0',
  'to' => '3.15.0-beta2',
)]
2023-07-10 09:13:25 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.14.0',
  'to' => '3.15.0-beta1',
)]
2023-07-10 09:13:26 [info] elementor::elementor_updater Started 
2023-07-10 09:13:26 [info] Elementor/Upgrades - _on_each_version Start  
2023-07-10 09:13:26 [info] Elementor/Upgrades - _on_each_version Finished 
2023-07-10 09:13:26 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.14.0',
  'to' => '3.15.0-beta2',
)]
2023-07-10 09:13:26 [info] elementor-pro::elementor_pro_updater Started 
2023-07-10 09:13:26 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-07-10 09:13:26 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-07-10 09:13:26 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.14.0',
  'to' => '3.15.0-beta1',
)]

== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	

				
</details>

**Comment by hein-obox** [comment]
This could be a bug related to our A11Y update.

**Comment by perks98** [comment]
@hein-obox I think you are right, from checking this #22738 and finding the frontend-modules JavaScript file to temporarily manually alter it seems to fix the issue.

This seems to be the area of interest:
![Screenshot_829](https://github.com/elementor/elementor/assets/55829113/8dbd7aee-41aa-4491-b264-4292ec4b3c90)

This is a bit of a bigger problem in certain use cases as a products carousel widget that I created that shows 4 products in view, it seems to add that `inert` on first load so users can interact with the product until it automatically slides.
<details>
<summary>Custom products carousel screenshot</summary>

![Screenshot_830](https://github.com/elementor/elementor/assets/55829113/3c0bbbca-4969-47d5-8a95-d23742f86a11)


</details>

<hr>

@hein-obox is this on your team's fix list?

**Comment by nicholaszein** [comment]
Hello @perks98! 

Thank you for submitting your issue! 🙏 

🏷️ We've labeled this issue and our team will take a look at it. If there is indeed a problem, they'll work on fixing it. We can't say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding!

Best regards

**Comment by perks98** [comment]
I doubt at this point they will fix it since it has been so long... They just need to remove the add and remove `inert` code they had added, this won't break websites by removing and will fix this issue then they can look in to a more stable solution later... but still nothing.

@nicholaszein any update? You put an internal reference of [ED-11434], so is there any movement?

**Comment by lw-development** [comment]
Hi all, I have been able to find a fix for this, hopefully this helps others!

This is a script which was added to a HTML element in a section above the carousel.

<script>
window.onload = function() {
    // Wait for the window to load

    // Find all elements with the class 'swiper-slide'
    var swiperSlides = document.querySelectorAll('.swiper-slide');

    // Loop through each swiper slide
    swiperSlides.forEach(function(slide) {
        // Check if the slide has the 'inert' attribute
        if (slide.hasAttribute('inert')) {
            // Remove the 'inert' attribute
            slide.removeAttribute('inert');
        }
    });
};
</script>

Thanks,
Louis

**Comment by adnanmukri** [comment]
Thanks Louis

I am going to check this.

**Comment by lw-development** [comment]
Thanks @adnanmukri, let me know how it goes!

**Comment by lw-development** [comment]
Pleased that it is helpful @eminozlem, hopefully we get a full fix soon!

**Comment by eminozlem** [comment]
> Pleased that it is helpful @eminozlem, hopefully we get a full fix soon!

Yea hopefully. Btw I ve used a slightly modified version of your code since 'onload' did not work for me i put a delay on it like so :
```

<script type="text/javascript">
window.addEventListener('load', function() {
        setTimeout(()=>{
// Find all elements with the class "swiper-slide"
var swiperSlides = document.querySelectorAll(".swiper-slide");

// Loop through each swiper slide
swiperSlides.forEach(function(slide) {

    // Check if the slide has the "inert" attribute
    if (slide.hasAttribute("inert")) {
        // Remove the "inert" attribute
        slide.removeAttribute("inert");
    }
});

        }, 1000)

  });
</script>
```

**Comment by lw-development** [comment]
> > Pleased that it is helpful @eminozlem, hopefully we get a full fix soon!
> 
> Yea hopefully. Btw I ve used a slightly modified version of your code since 'onload' did not work for me i put a delay on it like so :
> 
> ```
> 
> <script type="text/javascript">
> window.addEventListener('load', function() {
>         setTimeout(()=>{
> // Find all elements with the class "swiper-slide"
> var swiperSlides = document.querySelectorAll(".swiper-slide");
> 
> // Loop through each swiper slide
> swiperSlides.forEach(function(slide) {
> 
>     // Check if the slide has the "inert" attribute
>     if (slide.hasAttribute("inert")) {
>         // Remove the "inert" attribute
>         slide.removeAttribute("inert");
>     }
> });
> 
>         }, 1000)
> 
>   });
> </script>
> ```

That's great, thank you for the new snippet, it is really helpful!

**Comment by ervannur** [comment]
Hi, I found a workaround to remove the `inert` attribute using this code

```
const slider = $swiper.data('swiper');
document.querySelectorAll('[inert]').forEach((slide) => { slide.removeAttribute('inert') });
slider.off('slideChangeTransitionEnd', slider.params.on.slideChangeTransitionEnd);
```

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor v3.25.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
